server/controllers/watchedController.js

```diff
@@ -70,120 +67,63 @@ exports.update = async function(req, res) {
 
     watched.save()
     .catch((err) => {
-        res.status(400).send(err);
+        res.status(RequestStatus.BAD_REQUEST).send(err);
     })
     .then((updateWatched) => {
-        res.status(200).json(updateWatched);
+        res.status(RequestStatus.OK).json(updateWatched);
     });
 };
 
 exports.delete = async function(req, res) {
-    let watched_id = req.params.watched_id;
-
-    Watched.findById(watched_id)
-    .catch((err) => {
-        res.status(400).send(err);
-    })
-    .then((watched) => {
-        watched.remove()
-        .catch((err) => {
-            res.status(400).send(err);
-        })
-        .then((deletedWatched) => {
-            res.status(200).json(deletedWatched);
-        });
-    });
+    let watchedId = req.params.watched_id;
+    try {
+        await DataStoreUtils.deleteWatched(watchedId);
+        res.status(RequestStatus.OK);
+    } catch (err) {
+        res.status(RequestStatus.BAD_REQUEST).send(err);
+    }
 };
 
-var create_action = async function(user_id, watched_id) {
-    var action = new Action({
-        _user: user_id,
-        date: new Date(),
-        _action: watched_id,
-        action_type: WATCHED_ACTION_TYPE,
-    });
-    return action.save();
-}
-
-var delete_action = function(action_id) {
-    Action.remove({ _id: action_id}).exec();
-}
-
-var add_action_to_user_history = async function(user_id, action_id) {
-    User.findById(user_id, function (err, user) {
-        let user_history = user._history;
-        user_history.push(action_id);
-        return user.save();
-    });
-}
-
-var delete_action_from_user_history = async function(user_id, action_id) {
-    User.findById(user_id, function (err, user) {
-        let user_history = user._history;
-        let index = user_history.indexOf(action_id);
-        if (index > -1) {
-            user_history.splice(index, 1);
-        }
-        user.save();
-    });
-}
-
-var find_watched_obj = async function(watched_id) {
-    return Watched.findById(watched_id).exec();
-}
-
-var find_user_watched_list = async function(user_id) {
-    return Watched.find({_user: user_id}).exec();
-}
-
-
-getSeasonFromAPI = function(tv_id, season_number){
+var getSeasonFromAPI = function(tv_id, season_number){
   return new Promise(function(resolve, reject) {
-    var query = 'tvshow/' + tv_id + '/season/' + season_number
-    console.log("log")
+    var query = 'tvshow/' + tv_id + '/season/' + season_number;
     https.get("https://api.themoviedb.org/3/tv/"+ tv_id + "/season/"+ season_number +"?api_key=db00a671b1c278cd4fa362827dd02620",
     (resp) => {
       let data = '';
       resp.on('data', (chunk) => {
         data += chunk;
       });
       resp.on('end', () => {
-        console.log("saving season result to redis:"+  query)
+        console.log("saving season result to redis:"+  query);
         client.set(query, JSON.stringify(data));
```

Comment:
Client not defined,

Reply:
Esse método precisa ser refatorado. Ele foi copiado de tvShow, eu acho. Sendo que tvShow tem um client e watched. Se tu precisar agr, faz RedisClient.createAndAuth.

Reply:
Coloquei o RedisClient.createAndAuth e pegou. Mas faz sentido esse método ser refatorado pois ele é doido, mas eu não sei nem pra onde essas coisas do cache/api tão indo então não sei fazer isso.